source/NVDAObjects/window/winword.py

```diff
@@ -1430,10 +1466,520 @@ def _get_WinwordWindowObject(self):
+
+	def script_previousChartElement(self,gesture):
+		if self.keyIndex == 0:
+			self.keyIndex=self.elementsCount-1
+		else:
+			self.keyIndex=self.keyIndex-1
+		self.invokeChartElement(self.keyIndex)
+	script_previousChartElement.canPropagate=True
+
+	def script_activatePosition(self,gesture):
+		# Toggle browse mode pass-through.
+		self.passThrough = True
+		self.ignoreTreeInterceptorPassThrough=False
+		browseMode.reportPassThrough(self)
+	# Translators: Input help mode message for toggle focus and browse mode command in web browsing and other situations.
+	script_activatePosition.__doc__=_("Toggles between browse mode and focus mode. When in focus mode, keys will pass straight through to the application, allowing you to interact directly with a control. When in browse mode, you can navigate the document with the cursor, quick navigation keys, etc.")
+	script_activatePosition.category=inputCore.SCRCAT_BROWSEMODE
+
+	def script_disablePassThrough(self, gesture):
+		rangeStart=self.wordApplicationObject.ActiveDocument.Range().Start
+		self.wordApplicationObject.ActiveDocument.Range(rangeStart, rangeStart).Select()
+		eventHandler.executeEvent("gainFocus", api.getDesktopObject().objectWithFocus())
+
+	__gestures = {
+				"kb(laptop):upArrow":"nextChartElement",
+				"kb(desktop):upArrow":"nextChartElement",
+				"kb(laptop):downArrow":"previousChartElement",
+				"kb(desktop):downArrow":"previousChartElement",
+				"kb(desktop):leftArrow":"previousChartElement",
+				"kb(laptop):leftArrow":"previousChartElement",
+				"kb(desktop):rightarrow":"nextChartElement",
+				"kb(laptop):rightArrow":"nextChartElement",
+				"kb:enter": "activatePosition",
+				"kb(desktop):numpadEnter":"activatePosition",
+				"kb:space": "activatePosition",
+				"kb:escape": "disablePassThrough",
+	}
+
+class WordChartSeriesTrendline(WordChart):
+	_trendlineTypeMap = {_chartConstants.xlExponential: 'Exponential',
+								_chartConstants.xlLinear: 'Linear',
+								_chartConstants.xlLogarithmic: 'Logarithmic',
+								_chartConstants.xlMovingAvg: 'Moving Average',
+								_chartConstants.xlPolynomial: 'Polynomial',
+								_chartConstants.xlPower: 'Power' 
+	}
+
+	def __init__(self, windowHandle, wordApplicationObject, wordChartObject, keyIndex, seriesIndex, trendlineIndex=1):
+		self.windowHandle=windowHandle
+		self.wordApplicationObject=wordApplicationObject
+		self.seriesIndex=seriesIndex
+		self.trendlineIndex=trendlineIndex
+		self.trendlinesCount=wordChartObject.SeriesCollection(self.seriesIndex).Trendlines().Count
+		super(WordChartSeriesTrendline, self).__init__(windowHandle=windowHandle, wordApplicationObject=wordApplicationObject, wordChartObject=wordChartObject, keyIndex=keyIndex)
+
+	def _get_name(self):
+		if self.wordChartObject.SeriesCollection(self.seriesIndex).Trendlines(self.trendlineIndex).DisplayEquation or self.wordChartObject.SeriesCollection(self.seriesIndex).Trendlines(self.trendlineIndex).DisplayRSquared:

```

Comment:
Many times through out the code you write self.wordChartObject.SeriesCollection(self.seriesIndex).Trendlines(self.trendlineIndex)
This should be turned into a property getter on the trendlines NVDAObject, maybe called trendlinesItem or something. NVDA will cache this value per core cycle, thus greatly reducing the amount of COM calls needing to be made.